components/global/global_search_nav/global_search_nav.tsx

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import Flex from '@mattermost/compass-components/utilities/layout/Flex';
+
+import Search from 'components/search';
+import QuickSwitchModal from 'components/quick_switch_modal';
+import {
+    Constants,
+    ModalIdentifiers,
+    RHSStates,
+} from 'utils/constants';
+
+import * as Utils from 'utils/utils';
+
+type Props = {
+    rhsState: typeof RHSStates[keyof typeof RHSStates] | null;
+    rhsOpen: boolean;
+    isQuickSwitcherOpen?: boolean;
+    actions: {
+        showMentions: () => void;
+        openRHSSearch: () => void;
+        closeRightHandSide: () => void;
+        openModal: (modalData: {modalId: string; dialogType: any; dialogProps?: any}) => void;
+        closeModal: (modalId: string) => void;
+    };
+};
+
+const GlobalSearchNav = (props: Props): JSX.Element => {
+    const {rhsState, rhsOpen, actions: {closeRightHandSide, showMentions}} = props;
+
+    useEffect(() => {
+        document.addEventListener('keydown', handleShortcut);
+        document.addEventListener('keydown', handleQuickSwitchKeyPress);
+    }, []);
+
+    const searchMentions = () => {
+        if (rhsState === RHSStates.MENTION) {
+            closeRightHandSide();
+        } else {
+            showMentions();
+        }
+    };
+
+    const handleShortcut = (e: KeyboardEvent) => {
+        const {actions: {closeModal}} = props;
+
+        if (Utils.cmdOrCtrlPressed(e) && e.shiftKey) {
+            if (Utils.isKeyPressed(e, Constants.KeyCodes.M)) {
+                e.preventDefault();
+                closeModal(ModalIdentifiers.QUICK_SWITCH);
+                searchMentions();
+            }
+            if (Utils.isKeyPressed(e, Constants.KeyCodes.L)) {
+                // just close the modal if it's open, but let someone else handle the shortcut
+                closeModal(ModalIdentifiers.QUICK_SWITCH);
+            }
+        }
+    };
+
+    const toggleQuickSwitchModal = () => {
+        const {isQuickSwitcherOpen, actions: {openModal, closeModal}} = props;
+
+        if (isQuickSwitcherOpen) {
+            closeModal(ModalIdentifiers.QUICK_SWITCH);
+        } else {
+            openModal({
+                modalId: ModalIdentifiers.QUICK_SWITCH,
+                dialogType: QuickSwitchModal,
+            });
+        }
+    };
+
+    const handleQuickSwitchKeyPress = (e: KeyboardEvent) => {
```

Comment:
Is this really the place to handle this? Maybe we can locate this more appropriately.

Reply:
you mean is this component the place? I guess we can handle this in the parent but the quick switcher has its own events too so to unify those would be nice but I would say it should be done in another PR

Reply:
I agree that this doesn't belong to the Search Nav component as it's not functionally related.

@nevyangelova, do you know where this was implemented before? Functionally it belongs to the Channel Sidebar where the Quick Switcher UI lives and IMO it should stay there.

Reply:
No it is in the RHSSearchNav which is only rendered instead of the global header so for now it should be present in both. I can refactor it to be handled by channel sidebar but i would say we can open a ticket and do it after as this was just moving functionality.

Reply:
Interesting. Ya, can you open that ticket and then we can move on as is for now?

Reply:
https://mattermost.atlassian.net/browse/MM-37814